chain/processor.go

```diff
@@ -40,8 +38,8 @@ func (b *StatelessBlock) Execute(
 		sm        = b.vm.StateManager()
 		numTxs    = len(b.Txs)
 		t         = b.GetTimestamp()
-		cacheLock sync.RWMutex
-		cache     = make(map[string]*fetchData, numTxs)
+		fetchLock sync.RWMutex
```

Comment:
I'm not sure what this lock does? the fetcher could still be concurrently writing when you are reading from `f.Cache`?

Reply:
As mentioned below, `f.Cache` should be private (and access must be synchronized within `fetcher` to prevent panic)